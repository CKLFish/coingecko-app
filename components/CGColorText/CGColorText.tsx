import React from 'react';
import {Text, TextProps} from 'react-native';
import {useTheme} from 'react-native-paper';

import {TextUtil} from '../../Utils';

interface ColorTextProps extends Omit<TextProps, 'children'> {
  value: number;
  prefix?: string;
  postfix?: string;
}

const ColorText: React.FC<ColorTextProps> = props => {
  const {value, postfix = '%', prefix = '', style} = props;
  const {colors} = useTheme();

  return (
    <Text
      style={[
        {
          color:
            value > 0 ? colors.up : value === 0 ? colors.unchange : colors.down,
        },
        style,
      ]}>
      {`${prefix}${TextUtil.formatPercentage(value)}${postfix}`}
    </Text>
  );
};

export default ColorText;
