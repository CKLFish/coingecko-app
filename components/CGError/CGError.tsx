import React, {useCallback} from 'react';
import {useTranslation} from 'react-i18next';
import {StyleSheet, View} from 'react-native';
import {Button, Text} from 'react-native-paper';

type Props = {
  onPress?: () => void;
};

const CGError: React.FC<Props> = props => {
  const {onPress: propOnPress} = props;
  const {t} = useTranslation();

  const onPress = useCallback(() => {
    if (propOnPress) {
      propOnPress();
    }
  }, [propOnPress]);

  return (
    <View style={styles.container}>
      <View style={styles.inner}>
        <Text style={styles.centerText}>{t('NETWORK_FETCH_FAIL')}</Text>
        <Button onPress={onPress}>{t('NETWORK_FETCH_RETRY')}</Button>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexGrow: 1,
    flexDirection: 'row',
  },
  inner: {
    alignSelf: 'center',
    flex: 1,
  },
  centerText: {
    textAlign: 'center',
  },
});

export default CGError;
