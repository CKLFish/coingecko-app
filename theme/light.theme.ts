import {DefaultTheme} from 'react-native-paper';

const theme = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    // # For Up / Down / Unchanged
    up: 'green',
    down: 'red',
    unchange: 'grey',
    favourite: 'red',
  },
};

export default theme;
