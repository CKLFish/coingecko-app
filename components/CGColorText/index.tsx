import CGColorText from './CGColorText';

export default CGColorText;
