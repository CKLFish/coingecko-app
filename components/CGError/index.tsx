import CGError from './CGError';

export default CGError;
