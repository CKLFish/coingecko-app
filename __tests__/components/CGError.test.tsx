/**
 * @format
 */

import 'react-native';

import {shallow} from 'enzyme';
import React from 'react';
import {Provider} from 'react-native-paper';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import CGColorText from '../../components/CGColorText';
import CGError from '../../components/CGError';
import theme from '../../theme/light.theme';

describe('CGError', () => {
  test('withOutonPress', () => {
    const errorInstance = renderer.create(<CGError />);
    expect(errorInstance.toJSON()).toMatchSnapshot();
  });
  test('withOnPress', () => {
    const errorInstance = renderer.create(<CGError onPress={() => {}} />);
    expect(errorInstance.toJSON()).toMatchSnapshot();
  });
});
