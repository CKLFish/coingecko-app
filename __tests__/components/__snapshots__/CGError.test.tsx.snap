// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CGError withOnPress 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "flexGrow": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      NETWORK_FETCH_FAIL
    </Text>
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 0,
          "minWidth": 64,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            NETWORK_FETCH_RETRY
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`CGError withOutonPress 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "flexGrow": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      NETWORK_FETCH_FAIL
    </Text>
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 0,
          "minWidth": 64,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            NETWORK_FETCH_RETRY
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
